From: Alexander Graf <agraf@suse.de>
Date: Wed, 23 Nov 2011 02:26:14 +0100
Subject: [PATCH] ARM: Fix compile breakage for generic kernel
Patch-Mainline: No, will hopefully be fixed properly there

The default kernel broke for me with the following error:

arch/arm/kernel/head.o: In function `stext':
arch/arm/kernel/head.S:102: undefined reference to `PHYS_OFFSET'

This patch should fix this by defining PHYS_OFFSET in memory.h even when
in an assembly section.

Signed-off-by: Alexander Graf <agraf@suse.de>

diff --git a/arch/arm/include/asm/memory.h b/arch/arm/include/asm/memory.h
index a8997d7..2579e82 100644
--- a/arch/arm/include/asm/memory.h
+++ b/arch/arm/include/asm/memory.h
@@ -139,6 +139,16 @@
 #define page_to_phys(page)	(__pfn_to_phys(page_to_pfn(page)))
 #define phys_to_page(phys)	(pfn_to_page(__phys_to_pfn(phys)))
 
+#ifndef CONFIG_ARM_PATCH_PHYS_VIRT
+#ifndef PHYS_OFFSET
+#ifdef PLAT_PHYS_OFFSET
+#define PHYS_OFFSET	PLAT_PHYS_OFFSET
+#else
+#define PHYS_OFFSET	UL(CONFIG_PHYS_OFFSET)
+#endif
+#endif
+#endif
+
 #ifndef __ASSEMBLY__
 
 /*
@@ -186,14 +194,6 @@ static inline unsigned long __phys_to_virt(unsigned long x)
 #endif
 #endif /* __ASSEMBLY__ */
 
-#ifndef PHYS_OFFSET
-#ifdef PLAT_PHYS_OFFSET
-#define PHYS_OFFSET	PLAT_PHYS_OFFSET
-#else
-#define PHYS_OFFSET	UL(CONFIG_PHYS_OFFSET)
-#endif
-#endif
-
 /*
  * PFNs are used to describe any physical page; this means
  * PFN 0 == physical address 0.
